File: src/main/java/com/bytebreakstudios/tictactoe/condition/TieCondition.java
package com.bytebreakstudios.tictactoe.condition;

import com.bytebreakstudios.tictactoe.State;
import com.bytebreakstudios.tictactoe.action.TieAction;
import com.bytebreakstudios.board.Condition;

public class TieCondition extends Condition<State, TieAction> {
    public TieCondition() {
        super(new TieAction());
    }

    @Override
    public boolean check(State previous, State current) {
        return !previous.finished && !current.finished &&
                current.q != 0 &&
                current.w != 0 &&
                current.e != 0 &&
                current.a != 0 &&
                current.s != 0 &&
                current.d != 0 &&
                current.z != 0 &&
                current.x != 0 &&
                current.c != 0;
    }

    @Override
    public String description() {
        return null;
    }

    @Override
    public String serialize() {
        return null;
    }

    @Override
    public Condition deserialize(String s) {
        return null;
    }
}

File: src/main/scala/com/bytebreakstudios/tictactoe/scala/condition/TieCondition.scala
package com.bytebreakstudios.tictactoe.scala.condition

import com.bytebreakstudios.board.Condition
import com.bytebreakstudios.tictactoe.scala.State
import com.bytebreakstudios.tictactoe.scala.action.TieAction

object TieCondition extends Condition[State, TieAction](new TieAction) {

  override def check(prev: State, cur: State): Boolean = !prev.finished && !cur.finished && cur.q != 0 && cur.w != 0 && cur.e != 0 && cur.a != 0 && cur.s != 0 && cur.d != 0 && cur.z != 0 && cur.x != 0 && cur.c != 0

  override def description(): String = ???

  override def serialize(): String = ???

  override def deserialize(s: String) = ???
}